### ISSUE TITLE
Configure a root disk for every instance

### ISSUE THREAD

**Issue by cmd-ntrf** [ComputeCanada/magic_castle#11]
Currently, the root disk size of the instances depends on the flavor-image combination chosen. 

In OpenStack, it is possible for flavors to have a 0gb disk size. Starting with OpenStack Stein release, it will not be possible to create instance with flavor 0gb root disk without defining a volume. If we do not define a default root disk that is associated with a volume, the creation of instances will fail with next release of OpenStack.

Furthermore, with release before Stein, the absence of root disk size definition could cause issue for other vms sharing the same hypervisor.

The solution is to define a root disk size, just like we do for Azure and GCP.

**Comment by cmd-ntrf** [comment]
We can use the combination of Terraform OpenStack flavor data source and dynamic block to create a volume only if the chosen flavor does not define a root disk.

Here is an example:
```
provider "openstack" {}

locals {
  flavor = "p2-3gb"
}

data "openstack_images_image_v2" "image" {
  name = "CentOS-7-x64-2019-07"
}

data "openstack_compute_flavor_v2" "small" {
  name = local.flavor
}

resource "openstack_compute_instance_v2" "instance_1" {
  name            = "instance_1"
  flavor_name     = local.flavor
  key_pair        = "runcode_key"
  security_groups = ["default"]
  image_id        = data.openstack_images_image_v2.image.id

  dynamic "block_device" {
    for_each = data.openstack_compute_flavor_v2.small.disk > 0 ? [] : [{volume_size = 10}]
    content {
      uuid                  = data.openstack_images_image_v2.image.id
      source_type           = "image"
      destination_type      = "volume"
      boot_index            = 0
      delete_on_termination = true
      volume_size           = block_device.value.volume_size
    }
  }
}
```

**Comment by cmd-ntrf** [comment]
Since we are adding a root disk size parameter, we might as well use it for every cloud provider not only OpenStack. For AWS, Azure and GCP, we were inheriting the root disk size from the chosen image. We will explicitly state the size of the disk based on `var.root_disk_size`.

**Comment by cmd-ntrf** [comment]
This issue has been fixed in 5.2.
